3-UsingAPIs/Week1/assignment/ex1-johnWho.js

```diff
@@ -7,24 +7,30 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-export const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
 
-    const fullName = `${firstName} Doe`;
+export const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+        return;
+      }
 
-    callback(fullName);
-  }, 1000);
+      const fullName = `${firstName} Doe`;
+
+      resolve(fullName);
+    }, 1000);
+  });
+ 
 };
 
 function main() {
-  getAnonName('John', console.log);
+  getAnonName('John')
+  .then(console.log)
+  .catch(console.log);
```

Comment:
While your solution is fine, it looks like something is wrong with your VSCode setup, because it should automatically format your file such that the `.then()` and `.catch()` method are indented by two spaces:

```suggestion
  getAnonName('John')
    .then(console.log)
    .catch(console.log);
```